airflow/api_connexion/endpoints/auth_endpoint.py

```diff
@@ -0,0 +1,242 @@
+    resp = security_manager.oauth_remotes[provider].authorize_access_token()
+    if resp is None:
+        raise Unauthenticated(detail="You denied the request to sign in")
+    # Verify state
+    try:
+        jwt.decode(
+            state,
+            key=appbuilder.app.config['SECRET_KEY'],
+            algorithms='HS256',
+        )
+    except jwt.InvalidTokenError:
+        raise BadRequest(detail="State signature is not valid!")
+    log.debug("OAUTH Authorized resp: %s", resp)
+    # Retrieves specific user info from the provider
+    try:  # pylint: disable=too-many-nested-blocks
+        userinfo = security_manager.oauth_user_info(provider, resp)
+    except Exception as err:  # pylint: disable=broad-except
+        log.debug("Error retrieving user info. Error =>%s", err)
+        user = None
+    else:
+        log.debug("User info retrieved from %s: %s", provider, userinfo)
+        # User email is not whitelisted
+        if provider in security_manager.oauth_whitelists:
+            whitelist = security_manager.oauth_whitelists[provider]
+            allow = False
+            for e in whitelist:
+                if re.search(e, userinfo["email"]):
+                    allow = True
+                    break
+            if not allow:
+                raise Unauthenticated(detail="You are not authorized.")
+        else:
+            log.debug("No allow list for OAuth provider")
+        user = security_manager.auth_user_oauth(userinfo)
+        log.debug("User is %s", user)
+    if user is None:
+        raise Unauthenticated(detail="Invalid login")
+    login_user(user, remember=False)
+    return security_manager.create_tokens_and_dump(user, auth_schema)
+
+
+def auth_remoteuser():
+    """Handle remote user auth"""
+    appbuilder = current_app.appbuilder
+    security_manager = appbuilder.sm
+    if security_manager.auth_type != AUTH_REMOTE_USER:
+        raise Unauthenticated(detail="Authentication type does not match")
+    username = request.environ.get("REMOTE_USER")
+    if not username:
+        raise Unauthenticated(detail="Invalid login")
+    user = security_manager.auth_user_remote_user(username)
+    if user is None:
+        raise Unauthenticated(detail="Invalid login")
+    login_user(user, remember=False)
+    return security_manager.create_tokens_and_dump(user, auth_schema)
+
+
+@jwt_refresh_token_required
+def refresh_token():
+    """Refresh token"""
+    user = get_jwt_identity()
+    access_token = create_access_token(identity=user)
+    return {'access_token': access_token}
+
+
+@jwt_required
+def revoke_token():
+    """
+    An endpoint for revoking both access and refresh token.
+
+    This is intended for a case where a logged in user want to revoke
+    another user's tokens
+    """
+    body = request.json
+    try:
+        data = token_schema.load(body)
+    except ValidationError as err:
+        raise BadRequest(detail=str(err.messages))
+    token = decode_token(data['token'])
```

Comment:
Couldn't understand the logic here or I misinterpreted it here. How is a user going to provide the token of another user?